### PR TITLE
Build with LZ4 compression

### PR DESCRIPTION
**Pull request by constantinpape**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by constantinpape** [comment]
@conda-forge-admin, please rerender

**Comment by constantinpape** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ requirements:
     - bzip2
     - xz
     - zlib
+    - lz4
```
Comment (jakirkham) [inline recipe/meta.yaml]: I think you want [`lz4-c`]( https://github.com/conda-forge/lz4-c-feedstock ). The [`lz4`]( https://github.com/conda-forge/lz4-feedstock ) package includes Python bindings to the C library.

Comment (jakirkham) [reply recipe/meta.yaml]: Should add I suspect this is causing the build issue as the `run_exports` is not being used with the Python package.

Comment (constantinpape) [reply recipe/meta.yaml]: Thanks! I noted that I also forgot to add it to the run dependencies.

Comment (jakirkham) [reply recipe/meta.yaml]: I think `run_exports` should do that for you, but maybe we were getting a version of the `lz4-c` package that doesn't have `run_exports`?

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

**Comment by jakirkham** [comment]
There seems to be an issue with a missing Boost symbol. Not sure what is causing that.

```
undefined symbol: _ZN5boost10filesystem6detail18create_directoriesERKNS0_4pathEPNS_6system10error_codeE
```

ref: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=41376

**Comment by constantinpape** [comment]
> There seems to be an issue with a missing Boost symbol. Not sure what is causing that.

Ok, I will see if I can reproduce this locally with boost 1.70.

File: recipe/meta.yaml (hunk 1 above)
Comment (constantinpape) [reply recipe/meta.yaml]: I was not aware of the `run_exports` functionality, have read up on it and this makes sense.
I can probably leave out the runtime dependencies for the other compression libraries as well then. At least if they specify `run_exports`. I will check that later.

Comment (jakirkham) [reply recipe/meta.yaml]: AFAIK it doesn't cause any problems to list them as well.

Comment (constantinpape) [reply recipe/meta.yaml]: Ok, can't hurt to be more explicit. I will leave it in then.

**Comment by constantinpape** [comment]
I have the same issue when building with boost 1.70 locally (Ubuntu 18, gcc 7.3). I will see if it build with 1.69 etc. tomorrow. Maybe the diff between recipes for the boost versions is helpful.

**Comment by jakirkham** [comment]
Would it make sense to just pin to the last working version of Boost to get this in and then mess with upgrading Boost separately?

**Comment by constantinpape** [comment]
> Would it make sense to just pin to the last working version of Boost to get this in and then mess with upgrading Boost separately?

Makes sense. I pinned to 1.69, which works for me locally and will make an issue to keep track of 1.70 later.

**Comment by jakirkham** [comment]
Looks like it is running into some issues finding the LZ4 libraries on Windows. Perhaps CMake needs additional guidance here?

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

(To handle the fact that `boost-cpp` is now pinned.)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -20,22 +20,24 @@ requirements:
...
     - blosc
     - bzip2
+    - lz4-c
     - xz
     - zlib
     - nlohmann_json
   run:
     - python
-    - boost-cpp
+    - boost-cpp =1.69
```
Comment (jakirkham) [inline recipe/meta.yaml]:
```suggestion
    - boost-cpp 1.69.0
```

**Comment by constantinpape** [comment]
I think all issues are fixed (although in a bit of a hacky way, see #17).
However, the azure windows build is still failing, while the appveyor build went through.

@jakirkham Will conda-forge use the version built by appveyor if I merge this?

For reference, this is the azure traceback:
```
CMake Error at CMakeLists.txt:5 (project):
  Failed to run MSBuild command:

    MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 4.8.3761.0
    [Microsoft .NET Framework, version 4.0.30319.42000]
    Copyright (C) Microsoft Corporation. All rights reserved.
    
    Build started 8/5/2019 4:21:46 PM.
    Project "D:\bld\z5py_1565021703104\work\build\CMakeFiles\3.15.0\VCTargetsPath.vcxproj" on node 1 (default targets).
-- Configuring incomplete, errors occurred!
See also "D:/bld/z5py_1565021703104/work/build/CMakeFiles/CMakeOutput.log".
    D:\bld\z5py_1565021703104\work\build\CMakeFiles\3.15.0\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    Done Building Project "D:\bld\z5py_1565021703104\work\build\CMakeFiles\3.15.0\VCTargetsPath.vcxproj" (default targets) -- FAILED.
    
    Build FAILED.
    
    "D:\bld\z5py_1565021703104\work\build\CMakeFiles\3.15.0\VCTargetsPath.vcxproj" (default target) (1) ->
      D:\bld\z5py_1565021703104\work\build\CMakeFiles\3.15.0\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    
        0 Warning(s)
        1 Error(s)
    
    Time Elapsed 00:00:00.68
    

  Exit code: 1
```

**Comment by jakirkham** [comment]
The Windows Azure failures can be ignored. This is a known issue. We are still using AppVeyor for Windows builds for this reason, which appears to pass here.

**Comment by constantinpape** [comment]
Ok, got it.
Thanks for the help @jakirkham.

**Pull request merged**